galaxy_importer/loaders.py

```diff
@@ -72,6 +86,23 @@ def _make_name(self):
         """Returns content name generated from it's path."""
         pass
 
+    @abc.abstractmethod
+    def _get_path_name(self):
+        """Returns subdirectories as part content name.
+        'sub1.sub2.mod' for plugins/modules/sub1/sub2/mod.py"""
+        pass
+
+    def _get_fq_name(self):
+        return '{}.{}'.format(self.fq_collection_name, self.path_name)
+
+    def _get_tmp_dir(self):
+        root_parts = Path(self.root).parts
+        return os.path.join(*root_parts[:3])
+
+    def _get_fq_collection_name(self):
```

Comment:
If this and some of the other _get_* methods took args instead of using instance vars (`self.root` etc), these would be easy to unit test. 



``` python
    def _get_fq_collection_name(self, root):
          root_parts = Path(root).parts
          return '{}.{}'.format(*root_parts[-2:])
```

They could also potentially be either staticmethods or module scope methods as well. 

Reducing the number of methods that use instance variables likely allows getting rid of a lot of the instances vars which can simplify maintenance (less state on each object to track, less chance of unintentionally changing them etc).

For example, it looks like `self.doc_strings`, `self.readme_html` etc are mostly just used in load(). PluginLoader shouldn't need to keep a reference to 'doc_strings', it just needs to know how to create a schema.Content() in load() and return it.

Reply:
agreed and refactored